src/quo/components/wallet/confirmation_progress/view.cljs

```diff
@@ -0,0 +1,78 @@
+(ns quo.components.wallet.confirmation-progress.view
+  (:require [quo.components.wallet.confirmation-progress.style :as style]
+            [quo.components.wallet.progress-bar.view :as progress-box]
+            [quo.theme :as quo.theme]
+            [react-native.core :as rn]))
+
+(defn- calculate-box-state
+  [state counter index]
+  (cond
+    (and (= state :sending) (>= counter index) (< index 3))                 :confirmed
+    (and (= state :confirmed) (>= counter index) (< index 5))               :confirmed
+    (and (= state :finalising) (>= counter index) (< index 5))              :confirmed
+    (and (= state :finalising) (>= counter index) (> index 4) (< index 20)) :finalized
+    (and (= state :finalized) (>= counter index) (< index 5))               :confirmed
+    (and (= state :finalized) (>= counter index) (> index 4))               :finalized
+    (and (= state :error) (>= counter index) (< index 2))                   :error
+    :else                                                                   :pending))
+
+(defn- calculate-box-state-network-left
+  [state]
+  (cond
+    (= state :error)                                                     :error
+    (or (= state :confirmed) (= state :finalising) (= state :finalized)) :confirmed
+    :else                                                                :pending))
+
+(defn- calculate-box-state-network-right
+  [state]
+  (cond
+    (= state :error)
+    :error
+    (or (= state :confirmed)
+        (= state :finalising)
+        (= state :finalized))
+    :finalized
+    :else
+    :pending))
+
+(defn- calculate-progressed-value
+  [state progress-value]
+  (case state
+    :finalising progress-value
+    :finalized  100
+    0))
```

Comment:
I think you should put 0 & 100 in constants too - 
e.g  `(def ^:private max-progress 100)`, `(def ^:private min-progress 0)`

Reply:
Done